src/main/java/jnr/unixsocket/UnixSocketChannel.java

```diff
@@ -18,43 +19,46 @@
 
 package jnr.unixsocket;
 
+import java.io.IOException;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketOption;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.SocketChannel;
+import java.nio.channels.UnsupportedAddressTypeException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
 import jnr.constants.platform.Errno;
 import jnr.constants.platform.ProtocolFamily;
 import jnr.constants.platform.Sock;
 import jnr.constants.platform.SocketLevel;
-import jnr.constants.platform.SocketOption;
-import jnr.enxio.channels.NativeSocketChannel;
-import jnr.ffi.*;
-import jnr.ffi.byref.IntByReference;
-
-import java.io.IOException;
-import java.nio.channels.SelectionKey;
-import java.util.concurrent.locks.ReadWriteLock;
-import java.util.concurrent.locks.ReentrantReadWriteLock;
-
-import static jnr.unixsocket.SockAddrUnix.HEADER_LENGTH;
+import jnr.enxio.channels.AbstractNativeSocketChannel;
+import jnr.ffi.LastError;
 
 /**
- * A {@link java.nio.channels.Channel} implementation that uses a native unix socket
+ * A {@link java.nio.channels.Channel} implementation that uses a native unix
+ * socket
  */
-public class UnixSocketChannel extends NativeSocketChannel {
-    static enum State {
-        UNINITIALIZED,
-        CONNECTED,
-        IDLE,
-        CONNECTING,
-    }
-    private State state;
+public class UnixSocketChannel extends AbstractNativeSocketChannel {
+    enum State {
+        UNINITIALIZED, CONNECTED, IDLE, CONNECTING,
+    }
+
+    private volatile State state;
     private UnixSocketAddress remoteAddress = null;
     private UnixSocketAddress localAddress = null;
-    private final ReadWriteLock stateLock = new ReentrantReadWriteLock();
```

Comment:
@felfert Do you mean this? I think this was done by mistake while porting https://github.com/softprops/unisockets to java initially.